tests/components/qbus/test_switch.py

```diff
@@ -0,0 +1,97 @@
+"""Test Qbus switch entities."""
+
+from unittest.mock import MagicMock, patch
+
+import pytest
+from qbusmqttapi.const import KEY_OUTPUT_PROPERTIES, KEY_PROPERTIES_VALUE
+from qbusmqttapi.discovery import QbusMqttOutput
+from qbusmqttapi.state import QbusMqttOnOffState
+
+from homeassistant.components.qbus.switch import QbusSwitch
+from homeassistant.const import STATE_OFF, STATE_ON
+from homeassistant.core import HomeAssistant
+
+from .common import qbus_config_entry
+
+from tests.typing import MqttMockHAClient
+
+
+@pytest.mark.parametrize(
+    ("old_value", "new_value"),
+    [
+        (STATE_OFF, STATE_ON),
+        (STATE_ON, STATE_OFF),
+    ],
+)
+async def test_switch_state_received(
+    hass: HomeAssistant,
+    mqtt_mock: MqttMockHAClient,
+    qbus_switch_1: QbusSwitch,
```

Comment:
we should also avoid creating entitites directly and instead setting up the component normally and creating the entities automatically (in this case from MQTT messages)